### PR TITLE
Remove runner creation from every reload check

### PR DESCRIPTION
**Pull request by vjsamuel**
This PR will partly mitigate https://github.com/elastic/beats/issues/5139 as it removes runner creation during every reload interval.

### CODE REVIEW

**Comment by ruflin** [comment]
Travis build failure should be solved by: https://github.com/elastic/beats/pull/5142

File: libbeat/cfgfile/reload.go [hunk 1]
```diff
@@ -174,29 +175,34 @@ func (rl *Reloader) Run(runnerFactory RunnerFactory) {
 					continue
 				}
 
-				runner, err := runnerFactory.Create(c)
+				rawCfg := map[string]interface{}{}
```
Comment (ruflin) [inline libbeat/cfgfile/reload.go]: Not 100% sure anymore, but I think there is a difference between this and `ID()`. In the case of `ID()` the following to configs are the same:

```
a: b
c: d
```

and

```
c: d
a: b
```

That means the order does not matter, but with the above it does. But not 100% sure anymore if that was the reason I ended up with `ID()` but there was an issue.

**Comment by ruflin** [comment]
BTW: If we can remove the `ID` method 👍  from my side. I think also @urso did not like it when I introduced it but we didn't have a better way. That is why I worry the above breaks things.

File: libbeat/cfgfile/reload.go (hunk 1 above)
Comment (vjsamuel) [reply libbeat/cfgfile/reload.go]:
```
func main () {
	a := map[string]interface{}{
		"a": "b",
		"c": "d",
	}


	b := map[string]interface{}{
		"c": "d",
		"a": "b",
	}

	fmt.Println(hashstructure.Hash(a, nil))
	fmt.Println(hashstructure.Hash(b, nil))
}

```

output:
```
8710980024335696864 <nil>
8710980024335696864 <nil>

```

**Comment by vjsamuel** [comment]
@ruflin should `ID()` be removed in the same PR or a subsequent one?

**Comment by ruflin** [comment]
In case `ID()` is not used anywhere else it would be nice to directly remove it here to make sure we don't forget it. Could you rebase your PR on master because there was a build issue earlier today which is fixed now.

**Comment by vjsamuel** [comment]
@ruflin i will file a subsequent PR immediately later in the day to remove ID. i have to fix collectbeat as well. if you dont mind, i will create an issue and track it against me so that it is definitely taken care.

**Comment by exekias** [comment]
ok to test

**Comment by andrewkroh** [comment]
@ruflin @exekias Do we need to backport this to 5.6? Asking b/c of https://discuss.elastic.co/t/metricbeat-mongodb-leaks-connections-when-metricbeat-config-modules-reload-enabled-set-to-true/108723?u=andrewkroh

**Comment by ruflin** [comment]
@andrewkroh I think we could backport it, but as reloading was beta in 5.6 and is that 6.0 is out with the fix, I would rather encourage people to upgrade.

**Pull request merged**